source/core_sntp_client.c

```diff
@@ -515,17 +515,13 @@ static void rotateServerForNextTimeQuery( SntpContext_t * pContext )
 
 
 /**
- * @brief This function attempts to receive the SNTP response packet from a server
- * if the time window for server response has not timed out.
+ * @brief This function attempts to receive the SNTP response packet from a server.
  *
- * @note This function does not block on receiving the response packet from the network.
- * Instead, it determines whether the response packet is already available on the network
- * by first reading ONLY a single byte of data first.
- * If the single byte is available from the network, then rest of the SNTP response packet is
- * read from the network with retries for zero or partial reads until either:
- * - All the remaining bytes of server response are received
- *                     OR
- * - A timeout of #SNTP_RECV_POLLING_TIMEOUT_MS occurs of receiving no data over the network.
+ * @note This function treats reads of data sizes less than the expected server response packet,
+ * as an error as UDP does not support partial reads. Such a scenario can exist either due:
```

Comment:
It would be interesting to test this change with a low MTU and see if our default setup can handle IP fragmentation and reassembly.

Reply:
Yeah, that can be a good test, however, in that case, the responsibility of assembling fragmented UDP packet frames is for the application developer to take into account while writing the UDP transport interface that is supplied to the library.

Reply:
I looked into it further to determine whether there can be a case with the Network Time Security (NTS) authentication mechanism to generate NTP packets with size extending the data link layer MTU-enforced payload threshold. 
NTS does significantly extend the size of the NTP packet but the range of total NTP packets can be from 228 bytes to 1468 bytes (based on this [blog](https://blog.apnic.net/2019/11/08/network-time-security-new-ntp-authentication-mechanism/) by one of the persons who wrote implementation for protocol library; the official specification, [RFC 8915](https://datatracker.ietf.org/doc/html/rfc8915) does not specify a maximum total size.)  However, there is a way in the protocol to avoid certain extension fields (like the "NTS Cookie Placeholder Extension Field" that represents empty placeholders sent by the client to request more cookies from server in the response) to ensure that the total NTP packet size is kept under the MTU-enforced size.